tasks/scripting/src/main/java/org/infinispan/scripting/impl/ScriptingManagerImpl.java

```diff
@@ -230,10 +258,12 @@ <T> CompletionStage<T> execute(ScriptMetadata metadata, Bindings bindings) {
    }
 
    ScriptEngine getEngineForScript(ScriptMetadata metadata) {
-      ScriptEngine engine;
-      if (metadata.language().isPresent()) {
-         engine = scriptEnginesByLanguage.computeIfAbsent(metadata.language().get(), getEngineByName);
-      } else {
+      ScriptEngine engine = null;
+      Optional<String> language = metadata.language();
+      if (language.isPresent()) {
+         engine = scriptEnginesByLanguage.computeIfAbsent(language.get(), getEngineByName);
```

Comment:
Is it okay now if this returns a null engine?

Reply:
if engine is null we throw an exception further down